### PR TITLE
Feature/2160/alias redirect

### PR DESCRIPTION
**Pull request by agduncan94**
Removes previous alias code, now instead does the following:
* Added an aliases component at /aliases/type/alias, where `type` is in the set {organisations, collections} and `alias` is the alias
* If type is valid, will attempt to find the associated object for that alias using the appropriate alias endpoint
* If found object, redirect to page
* If not found, message saying no object with given alias found of some type

The benefit of this approach is that the organisations/collections code doesn't need to worry about aliases, they are all done in a separate component. Any new entity types (ex. entry) can be added to this component pretty easily.

The alias page shows the following, though this will only be visible for a split second, if that, unless on a slow connection.
redirect-alias

This PR also removes any links to organisations or collections using DB ids. **Now ONLY names will be used for these links.**

### CODE REVIEW

File: src/app/aliases/aliases.component.html [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<app-header>
+  Aliases
+</app-header>
+<div class="container">
+  <mat-card class="alert alert-warning" role="alert" *ngIf="!types.includes(type)">
```
Comment (garyluu) [inline src/app/aliases/aliases.component.html]: have a boolean variable for this ngIf

File: cypress/integration/group3/organizations.ts [hunk 2]
```diff
@@ -260,13 +260,72 @@ describe('Dockstore Organizations', () => {
...
+      cy.visit('/organizations/Potatoe/collections/veryFakeCollectionName');
       cy.title().should('eq', 'Dockstore | Collection');
     });
   });
+
+  describe('Find organization and collection by alias', () => {
+    it('organization alias', () => {
+      cy.server();
+      cy.route({
+        url: '/organizations/fakeAlias/aliases',
```
Comment (denis-yuen) [inline cypress/integration/group3/organizations.ts]: So it I understand a visit to `/aliases/organizations/fakeAlias` should redirect to `Cypress.config().baseUrl + '/organizations/Potatoe`. But what role is `/organizations/fakeAlias/aliases` playing here?

File: src/app/aliases/aliases.component.html [hunk 3]
```diff
@@ -0,0 +1,26 @@
+<app-header>
+  Aliases
```
Comment (denis-yuen) [inline src/app/aliases/aliases.component.html]: Could probably just call it "redirecting..." or something similar.

File: src/app/aliases/aliases.component.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    super();
+  }
+
+  ngOnInit() {
+    this.type = this.route.snapshot.paramMap.get('type');
+    this.alias = this.route.snapshot.paramMap.get('alias');
+
+    this.loading$ = this.aliasesQuery.loading$;
+
+    if (this.types.includes(this.type)) {
```
Comment (coverbeck) [inline src/app/aliases/aliases.component.ts]: I don't think this if is necessary, given the explicit tests you have for `organizations` and `collections` inside this if, but I could be missing something.

Comment (agduncan94) [reply src/app/aliases/aliases.component.ts]: I think that is a relic of something I was trying out before, I'll remove it.

File: cypress/integration/group3/organizations.ts (hunk 2 above)
Comment (agduncan94) [reply cypress/integration/group3/organizations.ts]: This is the API call being used to get an organizations alias called fakeAlias, this is not what is seen in the browser URL.

**Pull request merged**